### DISCUSSION TITLE
CSS In JS 관련 질문 입니다!

### DISCUSSION THREAD

**Discussion by baechanghyeon** [WantedPreonboardingFE5team/pre-onboarding-7th-2-1-5#44] [General]
이번에 TypeScript 를 쓰면서 CSS에 대해 한번 더 생각을 하게 됐습니다. 
그래서 다른  CSS 라이브러리를 찾아보면서 emotion에 대해 알게되어 한번 정리도 할겸해서 공유 할려고 합니다...! 

emotion에는 여러가지 스타일 주는 방법이 있지만 대표적인 3가지 방법을 들고 와봤습니다.

1. css prop

```jsx
import { css } from '@emotion/react'

const color = 'white'

render(
  <div
    css={css`
      padding: 32px;
      background-color: hotpink;
      font-size: 24px;
      border-radius: 4px;
      &:hover {
        color: ${color};
      }
    `}
  >
    Hover to change color.
  </div>
)
```

2. styled-components

```jsx
import styled from '@emotion/styled'

const Button = styled.button`
  padding: 32px;
  background-color: hotpink;
  font-size: 24px;
  border-radius: 4px;
  color: black;
  font-weight: bold;
  &:hover {
    color: white;
  }
`

render(<Button>This my button component.</Button>)
```

3. composition

```jsx
import { css } from '@emotion/react'

const danger = css`
  color: red;
`
const base = css`
  background-color: darkgreen;
  color: turquoise;
`
render(
  <div>
    <div css={base}>This will be turquoise</div>
    <div css={[danger, base]}>
      This will be also be turquoise since the base styles
      overwrite the danger styles.
    </div>
    <div css={[base, danger]}>This will be red</div>
  </div>
)
```

## emotion 
- styled-components와 같은 방법으로 스타일을 줄 수 있다.
- render 부분 확인 시 어떤 태그를 사용했는지 바로 알 수 있다. ( div, span, p, button, input )
- 이외에도 다양한 사용 방식이 존재한다. ( 대표적인 스타일 3가지 )
- 공식 URL :[Emotion 공식 주소](https://emotion.sh/docs/introduction)
- 참고 URL :[Emotion 방법 소개 블로그 주소](https://99geo.tistory.com/71)

이 때까지 프로젝트를 진행하시면서 어떤 CSS를 사용해보셨나요? 

혹시 사용하셨던 CSS에 대한 장점에 대해 다른 분들의 의견을 간단하게 들어보고 싶슴니다!

**Comment by jaeyeong815** [comment]
저는 styled-component만 써왔어요
가끔 MUI정도? 근데 MUI로는 그냥 스피너나 이런 요소를 주로 사용하게 되고 전반적으로는 styled-component를 썼네요
그래서 장점이라고 말할건 딱히 없는 것 같고 ,, css랑 비슷하게 쓰이니까 처음에 쉽게 접근할 수 있던 것 같아요!

emotion이 스타일드 컴포넌트랑 비슷지만 composition 부분을 보니 다른게 확 느껴지네요!
공식문서 훑고오겠습니닷

**Reply by baechanghyeon** [reply]
저도 CSS In JS 중에서는 styled-components 만 써봐서 이번에 과제 진행하면서 많은 라이브러리가 있다는걸 알게 됐습니다...ㅎㅎ
보다가 다른분들은 어떻게 생각하시는지 궁금해서 보니까 공통점이 있으면서도 딱히 이거다라는건 없는거 같긴 합니다..찍먹이 답!👊

**Reply by jaeyeong815** [reply]
ㅋㅋㅋ 이번 기회에 저희 여러가지를 한번 써봐요 ! 이럴 때 아니면 혼자서 써볼 기회가 없어서 ㅠ_ㅠ .. 혼자하면 자꾸 익숙한거 찾게돼요 하핫

**Reply by baechanghyeon** [reply]
ㅋㅋㅋㅋㅋㅋ저도 동감합니다... 어려운걸 보다보면 익숙한걸 찾게 되더라구요...ㅜㅜ

**Comment by yeondooo** [comment]
저는 전에도 말했지만 보통 tailwind css 많이 쓰는 편입니다. 이것도 다양한 기능이 많이 있는것 같지만..아직은 잘 활용 못하는것 같긴 한데 무엇보다 편해서 좋습니다. 변수명, 함수명 짓는게 제일 힘든데 그런 고민을 덜게 해준다는 이유도 좋구요! 
공식 사이트 URL: https://tailwindcss.com/
코딩애플의 영업 URL: https://youtu.be/--D4WMPEIZI
사용법 URL: https://wonny.space/writing/dev/hello-tailwind-css

여태 이것만 쓰다보니 다른 것도 써보고 싶다는 생각이 들었는데  조금씩 찍먹해보고 원하는 스타일을 찾는것도 좋겠습니다!

**Reply by baechanghyeon** [reply]
코딩애플 영업 영상 보고 저도 당할뻔했네요.. 
tailwind css는 제가 오늘 본 next.js 영상에서 사용하길래.. 보면서 뭐 저렇게 복잡해 했는데 어느정도 알면은 또 사용하기는 괜찮은가 보네요
말씀처럼 자신에게 맞는 CSS 그리고 다양한 경험이 답인거 같긴합니다.. 감사합니다!

**Reply by jaeyeong815** [reply]
저도 꼭 테일윈드 사용해보고싶네요~,~ 도전 해볼 기회가 와야할텐데 사전 과제였던 투두리스트 css를 이거로 한번 리팩토링 해봐야겠네용

**Reply by baechanghyeon** [reply]
tailwind 보니까 느낌이 bootstrap? 이랑 많이 비슷하다고 저는 느낌을 받았는데 어느정도 알게되면 또 tailwind만의 매력이 있을거 같습니다 👍
